src/fal/fal_script.py

```diff
@@ -23,12 +23,18 @@ def exec(self, context, faldbt: FalDbt):
 
         # Enable local imports
         local_path = str(self.path.parent)
-        sys.path.append(local_path)
+        try:
+            # NOTE: since this happens in threads, the `local_path` available
+            # in `sys.path` for all scripts running at the same time.
+            # This may introduce undesired race conditions for users.
+            # We probably want to pass the `sys.path` of each separately
+            sys.path.append(local_path)
+
+            with open(self.path) as file:
+                source_code = compile(file.read(), self.path, "exec")
 
-        with open(self.path) as file:
-            a_script = file.read()
             exec(
```

Comment:
is this the right indentation level?

Reply:
Yes, we get the `source_code` from the file and then close it right away, instead of keeping it open for `exec`